Public/Src/Engine/UnitTests/Scheduler/PipFingerprinterTests.cs

```diff
@@ -400,6 +401,94 @@ public void TestUserProfileStaysOutOfFingerprints()
 
+        private ContentFingerprint GetWeakFingerPrintForExecutable(
+            string exectuablePath, 
+            bool assertValue,  // true if fingerprint should contain executable path, false if output should be untracked
+            bool untracked = false, 
+            string untrackedScopePath = null)
+        {
+            var pathTable = m_context.PathTable;
+            var executable = FileArtifact.CreateSourceFile(AbsolutePath.Create(pathTable, X(exectuablePath)));
+
+            var dependencies = new HashSet<FileArtifact>() { executable };
+            var processBuilder = GetDefaultProcessBuilder(pathTable, executable)
+                .WithDependencies(dependencies);
+
+            if (untracked)
+            {
+                processBuilder = processBuilder.WithUntrackedPaths(new AbsolutePath[] { AbsolutePath.Create(pathTable, X(exectuablePath)) });
+            }
+
+            if (untrackedScopePath != null)
+            {
+                processBuilder = processBuilder.WithUntrackedScopes(new AbsolutePath[] { AbsolutePath.Create(pathTable, OperatingSystemHelper.IsUnixOS ? "/tmp" : Environment.GetFolderPath(Environment.SpecialFolder.History)) });
+            }
+
+            var process = processBuilder.Build();
+
+            MountPathExpander expander = new MountPathExpander(pathTable);
+
+            var fingerprinter = new PipContentFingerprinter(
+                m_context.PathTable,
+                GetContentHashLookup(executable),
+                ExtraFingerprintSalts.Default(),
+                pathExpander: expander)
+            {
+                FingerprintTextEnabled = true
+            };
+
+            var contentFingerprint = fingerprinter.ComputeWeakFingerprint(process, out string fingerprintText);
+            fingerprintText = fingerprintText.ToUpperInvariant();
+            string untrackedPathText = executable.Path.ToString(pathTable).ToUpperInvariant();
+
+            // Make sure the executable path doesn't show up in the fingerprint
+            XAssert.Equals(assertValue, fingerprintText.Contains(untrackedPathText));
+
+            // Validate the JSON fingerprinter does the same
+            var json = JsonFingerprinter.CreateJsonString(writer =>
+            {
+                fingerprinter.AddWeakFingerprint(writer, process);
+            },
+            pathTable: m_context.PathTable).ToUpperInvariant();
+
+            var fence = OperatingSystemHelper.IsUnixOS ? "\"" : "";
+            XAssert.Equals(assertValue, json.Contains($"{fence}{untrackedPathText.Replace("\\", "\\\\")}{fence}"));
+
+            return contentFingerprint;
+        }
+
+        [Fact]
+        public void TestExecutableWhenTrackedStaysInsideFingerprints()
+        {
+            var fingerprint = GetWeakFingerPrintForExecutable("/x/pkgs/tool.exe", true);
+            var fingerprint2 = GetWeakFingerPrintForExecutable("/x/pkgs/tool2.exe", true);
+            XAssert.IsFalse(fingerprint.Equals(fingerprint2));    

```

Comment:
Prefer `AreNotEqual` to `IsFalse` because the former will produce better message when the assertion doesn't hold
```suggestion
            XAssert.AreNotEqual(fingerprint, fingerprint2);
```